src/mono/mono/metadata/class-init.c

```diff
@@ -4189,6 +4196,89 @@ mono_class_set_runtime_vtable (MonoClass *klass, MonoVTable *vtable)
 	klass->runtime_vtable = vtable;
 }
 
+static int
+index_of_class (MonoClass *needle, MonoClass **haystack, int haystack_size) {
+	for (int i = 0; i < haystack_size; i++)
+		if (haystack[i] == needle)
+			return i;
+
+	return -1;
+}
+
+static void
+build_variance_search_table_inner (MonoClass *klass, MonoClass **buf, int buf_size, int *buf_count, MonoClass *current) {
```

Comment:
`klass` is unused, except to pass down in the recursive call.

Reply:
Thanks, that's rubbish left over from when I was precomputing the offsets (which isn't valid to do anymore)